media/windows/ECC-MQTT-IoT/ECC_MQTT_IoT_SQLite.py

```diff
@@ -101,16 +124,149 @@
+            print(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+            logger.error(F"GZIP thread cancelled")
+            logger.error(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+        elif fn.done():
+            error = fn.exception()
+            if error:
+                print(F"Error occurred while attempting to GZIP log file...")
+                print(F"...log file {result[1]}")
+                print(F"...error returned: {error}")
+                print(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+                logger.error(F"Error occurred while attempting to GZIP log file...")
+                logger.error(F"...log file {result[1]}")
+                logger.error(F"...error returned: {error}")
+                logger.error(F"...log file {result[1]} should be manually GZIP'd and the source deleted.")
+            else:
+                result = fn.result()
+                print(F"{result[0]}")
+                logger.info(F"{result[0]}")
+                done_flag = True
+                # GZIP successful, so remove the source log archive
+                try:
+                    # The destination file is returned as the second element of the result list
+                    os.remove(result[1])
+                    print(F"Archived log file {result[1]} has been deleted.")
+                    logger.info(F"Archived log file {result[1]} has been deleted.")
+                except OSError as e:
+                    print(F"Error occurred while attempting to delete source log after GZIP...")
+                    print(F"...file {result[1]}, error {e}")
+                    print(F"...This file should be deleted manually.")
+                    logger.error(F"Error occurred while attempting to delete source log after GZIP...")
+                    logger.error(F"...file {result[1]}")
+                    logger.error(F"...This file should be deleted manually.")
+
+
+# Set the number of backup copies of the log to maintain
+max_log_archives = args.max_log_archives    # default is 9 archives if not specified
+# Set the maximum size in bytes at which the log is archived
+max_log_size = args.archive_log_size  # default is 1GB if not specified
+
 # Set up logging...change as appropriate based on implementation location and logging level
 log_file_path = args.log_file_path
 # log_file_path = None  # To direct logging to console instead of file
 
+logformatter = logging.Formatter('%(asctime)s:%(levelname)s: %(name)s: line: %(lineno)d %(message)s')
+logRotateZip = GZipRotator(log_file_path,
+                           maxBytes=max_log_size,
+                           backupCount=max_log_archives,
+                           delay=False)
+logRotateZip.setLevel(logging.DEBUG)
+logRotateZip.setFormatter(logformatter)
+logger = logging.getLogger('MQTTIoT')
+logger.addHandler(logRotateZip)
+logger.setLevel(logging.DEBUG)
+
+"""
```

Comment:
You might as well delete this old code -- it'll be in Git history if we need it.